httpclient-jetty/src/main/java/io/fabric8/kubernetes/client/jetty/JettyHttpClient.java

```diff
@@ -76,19 +78,37 @@ public DerivedClientBuilder newBuilder() {
 
   @Override
   public CompletableFuture<HttpResponse<AsyncBody>> consumeLines(
-      HttpRequest originalRequest, BodyConsumer<String> consumer) {
-    throw new UnsupportedOperationException("Not supported by the Jetty client");
+      HttpRequest originalRequest, AsyncBody.Consumer<String> consumer) {
+    final var request = toStandardHttpRequest(originalRequest);
+    final var future = new JettyAsyncResponseListener(request) {
+
+      final StringBuilder builder = new StringBuilder();
+
+      @Override
+      protected void onContent(ByteBuffer content) throws Exception {
+        for (char c : StandardCharsets.UTF_8.decode(content).array()) {
```

Comment:
I wanted to add more tests regarding this, but I don't want to delay the merge of the 4430 PR.

The current implementation:
- supports only `\n` as line terminators/breaks
- ignores the last line if it's not terminated/breaks
- won't emit truncated messages (which I thought was part of the deal, TBH I didn't check anything and implemented from what I recalled from our conversations :sweat_smile:)

So how do you want me to proceed in the scope of this PR?

Reply:
> supports only \n as line terminators/breaks

Which if fine, that's what the other implementations are doing.

> ignores the last line if it's not terminated/breaks

The JDK implementation does this, okhttp currently does not - but did in 5.10.

> won't emit truncated messages (which I thought was part of the deal, TBH I didn't check anything and implemented from what I recalled from our conversations sweat_smile)

That is the desired behavior - it's just that it adds even more blocking to the current okhttp implementation.

> So how do you want me to proceed in the scope of this PR?

As long as we're comfortable with ignoring the possibility that multi-byte characters are split across messages (I'm willing to assume that), I would elevate this implementation and remove the okhttp and jdk ones.

Reply:
> As long as we're comfortable with ignoring the possibility that multi-byte characters

Depending on how the consumeLines is used downstream which I haven't checked :facepalm: and don't recall precisely.

> I would elevate this implementation and remove the okhttp and jdk ones.

So my proposal is that I tackle this in a follow-up PR and we move on with this one to not delay the Vert.x implementation

Reply:
> Depending on how the consumeLines is used downstream which I haven't checked facepalm and don't recall precisely.

It's only used for http watches, which are only used if web sockets do not seem supported.

> So my proposal is that I tackle this in a follow-up PR and we move on with this one to not delay the Vert.x implementation

Either way - that next pr will still need to go in before the vert.x implementation - as it would remove the need to implement a consumeLines method.

Reply:
> It's only used for http watches, which are only used if web sockets do not seem supported.

Ok, so it's not a critical thing, and probably the API server is serving complete chunks. I wouldn't complicat it more then.

> as it would remove the need to implement a consumeLines method.

Seems like a valid point

Maybe we can discuss in the call